test/boost/sstable_datafile_test.cc

```diff
@@ -3278,3 +3278,63 @@ SEASTAR_TEST_CASE(sstable_identifier_correctness) {
         BOOST_REQUIRE_EQUAL(sst->sstable_identifier()->uuid(), sst->generation().as_uuid());
     });
 }
+
+SEASTAR_TEST_CASE(test_non_full_and_empty_row_keys) {
+    return test_env::do_with_async([] (test_env& env) {
+        auto random_spec = tests::make_random_schema_specification(
+                get_name(),
+                std::uniform_int_distribution<size_t>(1, 4),
+                std::uniform_int_distribution<size_t>(1, 4),
+                std::uniform_int_distribution<size_t>(2, 8),
+                std::uniform_int_distribution<size_t>(2, 8));
```

Comment:
random_note: designated initializers would be nicer here

Reply:
Yes